File: update_fonts.py
import os
import re

def walk(dir):
    results = []
    for root, dirs, files in os.walk(dir):
        if 'node_modules' in root or '.next' in root or 'studio' in root or '.git' in root:
            continue
        for file in files:
            if file.endswith(('.tsx', '.ts', '.jsx', '.js')):
                if 'studio' not in file:
                    results.append(os.path.join(root, file))
    return results

files = walk('./src')
changed_files = 0

for file in files:
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()

    # match exact classes font-bold, font-extrabold, font-black
    if re.search(r'\b(font-bold|font-extrabold|font-black)\b', content):
        new_content = re.sub(r'\b(font-bold|font-extrabold|font-black)\b', 'font-semibold', content)
        with open(file, 'w', encoding='utf-8') as f:
            f.write(new_content)
        changed_files += 1

print(f"Updated fonts in {changed_files} files.")

File: update_fonts.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function (file) {
        if (file.includes('node_modules') || file.includes('.next') || file.includes('studio') || file.includes('.git')) {
            return;
        }
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            results = results.concat(walk(file));
        } else if (file.endsWith('.tsx') || file.endsWith('.ts') || file.endsWith('.jsx') || file.endsWith('.js')) {
            results.push(file);
        }
    });
    return results;
}

const files = walk('./src');
let changedFiles = 0;

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    const regex = /\b(font-bold|font-extrabold|font-black)\b/g;
    
    if (regex.test(content)) {
        const newContent = content.replace(regex, 'font-semibold');
        fs.writeFileSync(file, newContent);
        changedFiles++;
    }
});

console.log(`Updated fonts in ${changedFiles} files.`);
